packages/embark-utils/src/index.js

```diff
@@ -0,0 +1,12 @@
+
+class Utils {
+
+  static joinPath() {
```

Comment:
A more modern style would be:

```js
static joinPath(...args) {
  const path = require('path');
  return path.join(...args);
}
```

I think it makes great sense to extract utils into a package in the monorepo, and I'm not opposed to going with this PR after req'd changes.  But, similar to discussion around `core/fs`, webpack and babel can handle our needs with respect to abstracting modules that work in node but need something different in the browser. In other words, we probably don't need `joinPath` at all, could just replace all usage with built-in `path.join`.